pkg/controller/restore_controller.go

```diff
@@ -462,43 +468,92 @@ func mostRecentCompletedBackup(backups []*api.Backup) *api.Backup {
 	return nil
 }
 
-func (c *restoreController) fetchBackup(objectStore cloudprovider.ObjectStore, name string) (*api.Backup, error) {
-	backup, err := c.backupLister.Backups(c.namespace).Get(name)
-	if err == nil {
-		return backup, nil
+// fetchBackupInfo checks the backup lister for a backup that matches the given name. If it doesn't
+// find it, it tries to retrieve it from one of the backup storage locations.
+func (c *restoreController) fetchBackupInfo(backupName string, pluginManager plugin.Manager) (backupInfo, error) {
+	var info backupInfo
+	var err error
+	info.backup, err = c.backupLister.Backups(c.namespace).Get(backupName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return backupInfo{}, errors.WithStack(err)
+		}
+
+		logContext := c.logger.WithField("backupName", backupName)
+		logContext.Debug("Backup not found in backupLister, checking each backup location directly, starting with default...")
+		return c.fetchFromBackupStorage(backupName, pluginManager)
 	}
 
-	if !apierrors.IsNotFound(err) {
-		return nil, errors.WithStack(err)
+	location, err := c.backupLocationLister.BackupStorageLocations(c.namespace).Get(info.backup.Spec.StorageLocation)
+	if err != nil {
+		return backupInfo{}, errors.WithStack(err)
 	}
 
-	logContext := c.logger.WithField("backupName", name)
+	info.objectStore, err = getObjectStoreForLocation(location, pluginManager)
+	if err != nil {
+		return backupInfo{}, errors.Wrap(err, "error initializing object store")
+	}
+	info.bucketName = location.Spec.ObjectStorage.Bucket
+
+	return info, nil
+}
 
-	logContext.Debug("Backup not found in backupLister, checking object storage directly")
-	backup, err = c.getBackup(objectStore, c.bucket, name)
+// fetchFromBackupStorage checks each backup storage location, starting with the default,
+// looking for a backup that matches the given backup name.
+func (c *restoreController) fetchFromBackupStorage(backupName string, pluginManager plugin.Manager) (backupInfo, error) {
+	locations, err := c.backupLocationLister.BackupStorageLocations(c.namespace).List(labels.Everything())
 	if err != nil {
-		return nil, err
+		return backupInfo{}, err
+	}
+
+	idx := 1
+	orderedLocations := make([]*api.BackupStorageLocation, len(locations))
+	for i := 0; i < len(locations); i++ {
+		if locations[i].Name == c.defaultBackupLocation {
+			orderedLocations[0] = locations[i]
+		} else {
+			orderedLocations[idx] = locations[i]
+			idx++
+		}
 	}
 
-	// ResourceVersion needs to be cleared in order to create the object in the API
-	backup.ResourceVersion = ""
-	// Clear out the namespace too, just in case
-	backup.Namespace = ""
+	logContext := c.logger.WithField("backupName", backupName)
+	var info backupInfo
+	for _, location := range orderedLocations {
+		info.objectStore, err = getObjectStoreForLocation(location, pluginManager)
```

Comment:
within the `orderedLocations` loop, I'd start with `log := logContext.WithField("locationName", location.Name)` -- then you don't need to append it in each of the subsequent log statements within the loop (and it's a structured field)